src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,233 @@
+
+                        isToken = false;
+                        tokenBuffer = null;
+                        isEncode = true;
+                    }
+                }
+                else if (currentCharacter == '{')
+                {
+                    if (position < length && resourceString[position] == '{')  // Treat as escape character for {{
+                    {
+                        if (isToken)
+                        {
+                            tokenBuffer.Append("{{");
+                            position++;
+                        }
+                        else
+                        {
+                            position++;
+                            if (position < length &&
+                                (resourceString[position] > '0' && resourceString[position] < '9'))
+                            {
+                                tokenBuffer = new StringBuilder();
+                                tokenBuffer.Append("{{");
+                                isToken = true;
+                            }
+                            else
+                            {
+                                outputBuffer.Append("{");
+                            }
+                        }
+                    }
+                    else
+                    {
+                        tokenBuffer = new StringBuilder();
+                        tokenBuffer.Append("{");
+                        isToken = true;
+                    }
+                }
+                else
+                {
+                    AppendToBuffer(isToken, currentCharacter.ToString(), ref tokenBuffer, ref outputBuffer);
+                }
+            }
+
+            return outputBuffer.ToString();
+        }
+
+        /// <summary>
+        /// Encodes the argument.
+        /// </summary>
+        /// <param name="argument">The object to encode.</param>
+        /// <returns>The encoded object.</returns>
+        private object EncodeArgument([NotNull] object argument)
+        {
+            Debug.Assert(!(argument is HtmlString));
+
+            return _encoder.HtmlEncode(argument.ToString());
+        }
+
+        private void AppendToBuffer(
+            bool isToken,
+            string value,
+            ref StringBuilder tokenBuffer,
+            ref StringBuilder outputBuffer)
+        {
+            if (isToken)
+            {
+                tokenBuffer.Append(value);
+            }
+            else
+            {
+                outputBuffer.Append(value);
+            }
+        }
+
+        private void AppendToOutputBuffer(
+            bool isEncode,
+            object[] arguments,
+            StringBuilder tokenBuffer,
+            ref StringBuilder outputBuffer)
+        {
+            if (tokenBuffer != null && tokenBuffer.Length > 0)
+            {
+                if (isEncode)
+                {
+                    outputBuffer.Append(EncodeArgument(string.Format(tokenBuffer.ToString(), arguments)));
```

Comment:
Avoid repeating `string.Format(tokenBuffer.ToString(), arguments)`; put that into a variable (above line 222) to make this and line 228 easier to understand. Optionally go further (also mixing in previous comment) e.g.
``` c#
  var outputString = string.Format(tokenBuffer.ToString(), arguments);
  if (isEncode)
  {
    outputString = _encoder.HtmlEncode(outputString);
  }

  outputBuffer.Append(outputString);
```